### PR TITLE
Fix Informational Issues

### LINKED ISSUE #257: Fix 'informational' issues (audit recommendations)
There are some very minor things brought up in the audit that we might want to fix. The following checklist only includes the item ID; the description and details can be found by looking at the audit report.
- [x] TYP-01 Confusing Variable Naming
- [x] SYS-03 Unnecessary Variable
- [x] STA-02 Lack of Validation for `destination_adress` on `send_payment_to_address()`
- [x] STA-01  Code duplication
- [x] SRW-01 Usage of hard-coded string
- [x] SRC-01 Unused Methods and Storage
- [x] PRO-01 Unhandled Error
- [x] PAE-02 Usage of magic numbers
- [x] PAE-01 Unnecessary `Result<>` Return Type
- [x] LIT-04 Reduce using `unwrap()` and `expect()` in Production Codebase
- [x] LIT-03 Panic can happen between correlated storage modifications
- [x] LII-01 Inconsistent match expression
- [x] LIE-04 Mismatch in variable name and pallet
- [x] LIL-01 Dead Code in `execute_replace()`
- [x] IMP-01 Same Behavior Defined for different conditions
- [x] EXC-01 Missing information in logging message
- [x] CLI-03 Incorrect error type thrown
- [x] CLI-02 Typos
- [x] CLI-01 Confusion function naming
- [ ] #417
- [x] 4FC-04 Logic should be moved to a separate function - refactoring
- [x] 4FC-03 Unused errors
- [x] 4FC-02 Inconsistent comments

Since the changes are so minor, it's okay to include all of them in one branch and one Pull Request. But each of them should at least be addressed in a separate commit with a proper name, also mentioning the **ID** in the name so that it's easier to track the changes.

additional:

- [ ] GLOBAL-05 Unneceessary Off-Chain User Protection Mechanism
- [x] LI5-02 TryFrom `CurrencyId` implementations Contain Repeated Code
- [x] LIH-03 Values Length Not Validated in `feed_values` Function

### PR DESCRIPTION
**Pull request by b-yap**
closes for #257 

This PR addresses the ff. items:
* ## SYT-03 | Unnecessary Variable
    * ### **from audit:**
       * The linked variable account_id is only used once in the function. Thus, its value can be supplied directly instead of stored in a variable.
    * ### **files covered:** 
      * system.rs
* ## SRL-01 | Usage of Hard-coded Strings
   * ### **from audit:**  
     * It is a good practice to declare and use Rust constants for such cases to better track their usage and avoid mistakes and bugs that may arise during several different development iterations.
   * ### **files covered:**
      * primitives/src/lib.rs
      * clients/
         * vault/tests/helper/constants.rs
         * wallet/src/
             * stellar_wallet.rs
             * horizon/responses.rs
* ## PAL-03 | Usage of Magic Numbers
  *  ### **from audit:**
     * We advise the team to declare constants to improve code maintainability and readability. E.g. the client could declare: HOURS_DURING_DAY , MINUTES_IN_HOUR , SECONDS_IN_HOUR, SECONDS_DURING_DAY, etc.
    * ### **files covered:**
      * pallets/
        * issue/src/lib.rs
        * redeem/src/lib.rs
* ## PAL-02 | Unnecessary Result<...> Return Type
  * ### **from audit:**
    * methods return Result<..., DispatchError> but that Error will never be thrown.
  * ### **files covered:**
    * pallets/
      * replace/src/
        * types.rs
        * lib.rs
      * redeem/src/
        * types.rs
        * lib.rs  
* ## CLI-02 |  Typos
  * ### **from audit:**
    * "insertin" should be "inserting".
    * "Oracle returned error" should be "Oracle returned an error".
    * "Not fetching missing envelopes from archive for slot {:?}, because on testnet" should be "Not fetching missing 
    * envelopes from archive for slot {:?}, because it is on testnet"
    * "the slot where the txset is to get" should be "the slot from where we get the txset"   
  * ### **files covered:**
    * clients/
      * runtime/src/shutdown.rs
      * vault/src/oracle/collector
        * collector.rs
        * proof_builder.rs
* ## 9B2-03 | Unused Errors
  * ### **from audit:** 
     * The errors HttpPostError and SeqNoParsingError are defined in vault/src/error.rs but are not used inside such crate.
     * The same thing happens for the errors NoOrganizationsRegisteredForNetwork, NoValidatorsRegisteredForNetwork, InvalidTransactionSet and InvalidTransactionXDR from stellar-relay/src/lib.rs.
  * ### **files covered:** 
     * clients/vault/src/error.rs
     * pallets/stellar-relay-src/lib.rs
* ## 9B2-04 | Logic Should be Moved To an Separate Function - Refactoring (partial)
  * ## **from audit:**
    * The function execute_open_requests() has over 150 LOC and does different things at once. This approach is not recommended as it increases the complexity of the whole function, may introduce errors and decreases the code's readability.
    * The last portion of the function is in charge of executing all the remaining requests on the Hashmap. As this functionality can be done in isolation, it is advised that it is moved to a separate function that receives the hashmap or list of open transactions pending execution.
  * ## **files covered:**
    * the whole _clients/vault/src/requests directory_, which replaces execution.rs
    * took the initiative to breakdown _vault_integration_tests.rs_ as well:
      * _see clients/vault/tests/helper_ directory 
* ## STL-03 | Lack of Validation for `destination_address` on `send_payment_to_address()` (partial)
  * ### **from audit:** 
    * The function send_payment_to_address() from stellar_wallet doesn't check if the destination address is the same user. Although this would not lead to malicious behavior, the action doesn't make sense and would make the user lose money in the fees involved in the transfer.
  * ### **files covered:**
    * clients/wallet/src/stellar_wallet.rs 
       *  see my [comment](https://github.com/pendulum-chain/spacewalk/pull/385#discussion_r1301198109) on why I exclude the checking for feature `testing-utils`.

additional PRs:
* #388 
* #390 
* #393 
    * #414

### CODE REVIEW

File: clients/vault/src/lib.rs [hunk 1]
```diff
@@ -21,13 +20,13 @@ mod system;
 
 mod issue;
 pub mod oracle;
+mod requests;
```
Comment (b-yap) [inline clients/vault/src/lib.rs:23]: `mod requests` replaces `mod execution`. It splits the old execution.rs into 3: 
* execution.rs
* helper.rs
* structs.rs

File: clients/wallet/src/horizon/responses.rs [hunk 2]
```diff
@@ -152,14 +193,45 @@ pub struct TransactionResponse {
 	pub memo: Option<Vec<u8>>,
 }
 
+impl Debug for TransactionResponse {
```
Comment (b-yap) [inline clients/wallet/src/horizon/responses.rs:196]: the simple derive `debug` prints the value of the fields in `Vec<u8>`. Converting them to a readable string will make it easier when we want to view the xdr.

File: clients/wallet/src/stellar_wallet.rs [hunk 3]
```diff
@@ -337,6 +337,12 @@ impl StellarWallet {
 		request_id: [u8; 32],
 		stroop_fee_per_operation: u32,
 	) -> Result<TransactionResponse, Error> {
+		// user must not send to self
+		#[cfg(not(feature = "testing-utils"))]
```
Comment (b-yap) [inline clients/wallet/src/stellar_wallet.rs:341]: right now the vault integration tests contains lots of self payment. 
This cfg guard is for avoiding CI issues, until the tests are updated.

File: clients/vault/Cargo.toml [hunk 4]
```diff
@@ -6,7 +6,9 @@ name = "vault"
 version = "0.0.1"
 
 [features]
-integration = []
+integration = [
+    "wallet/testing-utils"
```
Comment (b-yap) [inline clients/vault/Cargo.toml:10]: see `fn listen_for_new_transactions(...)` in _clients/wallet/src/horizon/horizon.rs_

File: clients/wallet/src/horizon/horizon.rs [hunk 5]
```diff
@@ -289,6 +289,7 @@ pub async fn listen_for_new_transactions<T, U, Filter>(
 	issue_map: Arc<RwLock<T>>,
 	memos_to_issue_ids: Arc<RwLock<U>>,
 	filter: Filter,
+	#[cfg(any(test, feature = "testing-utils"))] last_cursor: PagingToken,
```
Comment (b-yap) [inline clients/wallet/src/horizon/horizon.rs:292]: this is to address the "could not find event" errors. 
The `testing-utils` adds an extra param for a function that can define the "last known cursor".

See its usage in clients/vault/tests/vault_integration_tests.rs

File: clients/vault/src/requests/execution.rs [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+	)
+	.await?;
+
+	let rate_limiter = Arc::new(RateLimiter::direct(YIELD_RATE));
+
+	// Check if the open requests have a corresponding payment on Stellar
+	// and are just waiting to be executed on the parachain
+	spawn_tasks_to_execute_open_requests_async(
+		&mut open_requests,
+		wallet,
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/vault/src/requests/execution.rs:307]: good idea. I would prefer a new issue ticket for this.

File: clients/wallet/src/horizon/responses.rs [hunk 7]
```diff
@@ -23,20 +39,22 @@ pub(crate) async fn interpret_response<T: DeserializeOwned>(
 
 	let resp = response.json::<serde_json::Value>().await.map_err(Error::HttpFetchingError)?;
 
-	let unknown = "unknown";
-	let title = resp["title"].as_str().unwrap_or(unknown);
-	let status = u16::try_from(resp["status"].as_u64().unwrap_or(400)).unwrap_or(400);
+	let title = resp[RESPONSE_FIELD_TITLE].as_str().unwrap_or(VALUE_UNKNOWN);
+	let status =
+		u16::try_from(resp[RESPONSE_FIELD_STATUS].as_u64().unwrap_or(ERROR_STATUS_400 as u64))
+			.unwrap_or(ERROR_STATUS_400);
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/wallet/src/horizon/responses.rs]: unfortunately json can only convert strings to `u64`, and the http library's statuscode type is `u16` ( happens when we check for server errors. clients/wallet/src/error.rs `pub fn is_server_error(&self)` )

I introduced a `StatusCode` alias, to explicitly show the need to convert to u16.

File: clients/wallet/src/horizon/responses.rs [hunk 8]
```diff
@@ -23,20 +39,22 @@ pub(crate) async fn interpret_response<T: DeserializeOwned>(
...
 
 	let error = match status {
-		400 => {
-			let envelope_xdr = resp["extras"]["envelope_xdr"].as_str().unwrap_or(unknown);
+		ERROR_STATUS_400 => {
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/wallet/src/horizon/responses.rs]: used 400 back. 🙈

File: clients/vault/src/requests/execution.rs [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+	rate_limiter: Arc<RateLimiter<NotKeyed, S, C, MW>>,
+) where
+	S: DirectStateStore + Send + Sync + 'static,
+	C: ReasonablyRealtime + Send + Sync + 'static,
+	MW: RateLimitingMiddleware<C::Instant, NegativeOutcome = NotUntil<C::Instant>>
+		+ Send
+		+ Sync
+		+ 'static,
+	<MW as RateLimitingMiddleware<<C as Clock>::Instant>>::PositiveOutcome: Send,
+{
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/vault/src/requests/execution.rs:205]: I grouped them in a trait called `PayAndExecuteExt`, found in helper.rs. 
The struct `PayAndExecute` is in struct.rs

The implementation though I purposely did it in execution.rs. Seems fitting that it belongs with the other execution logic.

File: clients/vault/src/requests/helper.rs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+	wallet: Arc<RwLock<StellarWallet>>,
+) -> Option<TransactionsResponseIter> {
+	// Queries all known transactions for the targeted vault account and check if any of
+	// them is targeted.
+	let wallet = wallet.read().await;
+	let transactions_result = wallet.get_all_transactions_iter().await;
+	drop(wallet);
+
+	// Check if some of the requests that are open already have a corresponding payment on Stellar
+	// and are just waiting to be executed on the parachain
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/vault/src/requests/helper.rs]: removed

[the remark this answers is not in the document]
Comment (b-yap) [reply clients/vault/src/requests/helper.rs]: updated

File: clients/vault/tests/helper/mod.rs [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+		&self,
+		vault_id: &VaultId,
+		collateral: u128,
+		public_key: StellarPublicKey,
+	) -> Result<(), runtime::Error> {
+		self.register_public_key(public_key).await.unwrap();
+		self.register_vault(vault_id, collateral).await.unwrap();
+		Ok(())
+	}
+}
```
[the remark this answers is not in the document]
Comment (b-yap) [reply clients/vault/tests/helper/mod.rs]: done

File: clients/vault/src/requests/execution.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ebma) [reply clients/vault/src/requests/execution.rs:307]: Good point. I'll create a follow-up

File: clients/wallet/src/stellar_wallet.rs (hunk 3 above)
Comment (b-yap) [reply clients/wallet/src/stellar_wallet.rs]: updated: https://github.com/pendulum-chain/spacewalk/pull/385/commits/d3d2cc1e472c127bcc74cde62d220a34be6a5147

File: clients/vault/src/requests/execution.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ebma) [reply clients/vault/src/requests/execution.rs:307]: Created the follow-up [here](https://github.com/pendulum-chain/spacewalk/issues/416).

**Review by ebma — APPROVED** [review]
LGTM. We really need to create a follow-up ticket to clean up the code that is commented out though

**Pull request merged**